### PR TITLE
[podsecurity] Dedupe overlapping forbidden messages

### LINKED ISSUE #106129: [PodSecurity] Dedupe overlapping forbidden messages
Examples:
* restricted volumes complains about hostPath volumes already caught by baseline hostPath check
* restricted capabilities complains about added non-default capabilities already caught by baseline capabilities check
* restricted seccomp complains about unconfined seccomp profile already caught by baseline seccomp check

### PR DESCRIPTION
**Pull request by calvin0327**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Dedupe overlapping forbidden messages.

#### Which issue(s) this PR fixes:

Fixes #106129

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by calvin0327** [comment]
I have thought for a long time and added varible `overlap` in the struct of check. I think if we have more levels in the future, it will be easier to face these situations.

**Comment by calvin0327** [comment]
@liggitt Please correct me if I'm wrong.

File: staging/src/k8s.io/pod-security-admission/policy/registry.go [hunk 1]
```diff
@@ -69,6 +70,9 @@ func (r *checkRegistry) EvaluatePod(lv api.LevelVersion, podMetadata *metav1.Obj
 		results = append(results, check(podMetadata, podSpec))
 	}
 	if lv.Level == api.LevelBaseline {
+		for _, check := range r.overlappingBaselineChecks[lv.Version] {
```
Comment (liggitt) [inline staging/src/k8s.io/pod-security-admission/policy/registry.go:73]: marking checks as overlapping and skipping them without positive verification that the specific check they overlap with is also getting run makes me slightly nervous

the integration test failures are related, and make it seem like some expected checks aren't getting run

**Comment by tallclair** [comment]
@calvin0327 thank you for looking into this. After giving it some more thought, I'd like to propose an alternative approach, which is implemented in https://github.com/kubernetes/kubernetes/pull/107698. That proposes a larger change, but has stronger guarantees that the overridden checks are only skipped when the overriding check is present. It also allows for versioned overrides, in case a baseline check predates an overriding restricted check.

**Comment by calvin0327** [comment]
@tallclair👌，I think this implementation  #107698  is great.

**Pull request closed without merging**